index.js

```diff
@@ -21,27 +30,128 @@ app.use(cookieParser("secret"));
+            if(data.room in gameLogic.games) {
+                var game = gameLogic.games[data.room];
+                if(typeof game.player2 != "undefined") return;
+                if(game.player1.email === socket.handshake.query.email) {
+                    socket.emit("not_allowed");
+                    return;
+                }
+
+                socket.join(data.room);
+                rooms.push(data.room);
+                socket.room = data.room;
+                socket.pid = 2;
+                socket.hash = generateHash(8);
+                socket.username = socket.handshake.query.username;
+                socket.email = socket.handshake.query.email;
+                game.player2 = socket;
+                socket.opponent = game.player1;
+                game.player1.opponent = socket;
+                socket.emit('assign', {pid: socket.pid, hash: socket.hash});
+                game.turn = 1;
+                socket.broadcast.to(data.room).emit("start");
+                console.log(game.player1.username);
+                io.in(socket.room).emit("assign_names", { player1: game.player1.username, player2: game.player2.username });
+                console.log("\nPlayer 2 has joined room: ", socket.room);
+            } else {
+                if(rooms.indexOf(data.room) <= 0) socket.join(data.room);
+                socket.room = data.room;
+                socket.pid = 1;
+                socket.hash = generateHash(8);
+                socket.username = socket.handshake.query.username;
+                socket.email = socket.handshake.query.email;
+                gameLogic.games[data.room] = {
+                    player1: socket,
+                    moves: 0,
+                    board: [[0,0,0,0,0,0,0],
+                            [0,0,0,0,0,0,0],
+                            [0,0,0,0,0,0,0],
+                            [0,0,0,0,0,0,0],
+                            [0,0,0,0,0,0,0],
+                            [0,0,0,0,0,0,0]]
+                };
+                rooms.push(data.room);
+                socket.emit("assign", {pid: socket.pid, hash: socket.hash});
+                console.log("\nPlayer 1 has joined room: ", socket.room);
+            }
+    
+            socket.on('makeMove', function(data){
+                var game = gameLogic.games[socket.room];
+                if(data.hash = socket.hash && game.turn == socket.pid){
+                    var move_made = gameLogic.make_move(socket.room, data.col, socket.pid);
+
+                    if(move_made){
+                        game.moves = parseInt(game.moves)+1;
+                        socket.broadcast.to(socket.room).emit('move_made', {pid: socket.pid, col: data.col});
+                        game.turn = socket.opponent.pid;
+                        var winner = gameLogic.check_for_win(game.board);
+
+                        if(winner) {
+                            io.in(socket.room).emit("winner", {winner: winner});
+                        }

```

Comment:
@jkk2000 winner found here, match with email in DB and increment number of wins. before that create new field in schema - `no_of_wins` and default to zero.